packages/language-server/src/plugins/typescript/features/CompletionProvider.ts

```diff
@@ -103,9 +103,11 @@ export class CompletionsProviderImpl implements CompletionsProvider<CompletionEn
             return null;
         }
 
-        const { lang, tsDoc, userPreferences } = await this.lsAndTsDocResolver.getLSAndTSDoc(
-            document
-        );
+        const {
+            lang: langForSyntheticOperations,
+            tsDoc,
+            userPreferences
+        } = await this.lsAndTsDocResolver.getLsForSyntheticOperations(document);
```

Comment:
Reuse completion check and JSDoc template both don't need the language service program to be up to date.